stdlib/public/runtime/CMakeLists.txt

```diff
@@ -103,16 +103,39 @@ if(SWIFT_BUILD_STATIC_STDLIB AND "${sdk}" STREQUAL "LINUX")
   string(TOLOWER "${sdk}" lowercase_sdk)
 
   # These two libraries are only used with the static swiftcore
-  add_swift_library(swiftImageInspectionShared STATIC
+  add_swift_library(swiftImageInspectionShared TARGET_LIBRARY STATIC
     ImageInspectionELF.cpp
     C_COMPILE_FLAGS ${swift_runtime_library_compile_flags}
-    LINK_FLAGS ${swift_runtime_linker_flags})
-  set_target_properties(swiftImageInspectionShared PROPERTIES
-    ARCHIVE_OUTPUT_DIRECTORY "${SWIFTSTATICLIB_DIR}/${lowercase_sdk}")
+    LINK_FLAGS ${swift_runtime_linker_flags}
+    INSTALL_IN_COMPONENT stdlib)
+
+  foreach(arch IN LISTS SWIFT_SDK_${sdk}_ARCHITECTURES)
+    add_custom_command_target(swift_image_inspection_${arch}_static
+      COMMAND
+        "${CMAKE_COMMAND}" -E copy
+        "${SWIFT_BINARY_DIR}/lib/swift/${lowercase_sdk}/${arch}/libswiftImageInspectionShared.a"
```

Comment:
Can you see if you can get away with a generator expression?

    set(FragileSupportLibrary swiftImageInspectionShared-${SWIFT_SDK_${sdk}_LIB_SUBDIR}-${arch})
    set(LibraryLocation ${SWIFTSTATICLIBRARY_DIR}/${lowercase_sdk}/${arch})

    add_custom_command_target(swift_image_inspection_${arch}_static
                              COMMAND
                                "${CMAKE_COMMAND} -E copy $<TARGET_FILE:${FragileSupportLibrary}> ${LibraryLocation}
    ...